meta-resin-common/recipes-support/hostapp-update-hooks/files/0-bootfiles

```diff
@@ -90,16 +104,31 @@ deploy() {
 # MAIN
 #
 
-export DOCKER_HOST="unix:///var/run/docker-host.sock"
-
-if [ -z "$container_id" ]; then
-	echo "[ERROR] Container ID provided as an argument is needed."
-	exit 1
-fi
+# Do a dry run for copying the boot files and figure out if we would get in an
+# out of space situation
+boot_space="$(df -B1 --output=avail $boot_mountpoint | grep -v Avail)"
+available="$boot_space"
+available_threshold="524288" # All sizes in bytes
+printf "[INFO] Checking if boot partition can accommodate the new update... "
+for filename in $(find /resin-boot -type f | sed 's#/resin-boot##g'); do
```

Comment:
This is fragile. Consider changing to something like this:

```bash
for p in "/resin-boot/"*; do
    local _filename=$(basename "$p")
    ....
done

Reply:
This is not working as I needed because:
1. that for is not recursively
2. I need relative paths to /resin-boot not filenames

Reply:
Was wondering if the two for loops could work from the same list, i.e. finding the files ones, and reusing the results?

```
filelist=($(find ....))
for filename in "$filelist[@]"; do...
```
Depends on Petros' comment above, though if using find/sed is too fragile. Also in general would use `'s#^/resin-boot##g'` in any case, as to match that only at the beginning of the path, not anywhere in the path.

One more, if it's a relative path that you are looking for, the `filename` name is a bit misleading in expectations, IMHO.

Reply:
Reusing the loops as you propose would introduce bashisms so I'll let the current implementation. The next two comments though will be included in the next interation. Thanks.